### PR TITLE
fix(spinner): remove basespinner

### LINKED ISSUE #1280: Remove BaseSpinner Dependency
See https://github.com/patternfly/patternfly-elements/pull/2627

### PR DESCRIPTION
**Pull request by bennypowers**
Closes #1280 
## What I did

1. remove basespinner
2. deprecate colorPalette on spinner (cc @marionnegp )

## Testing Instructions

1. compare dp to prod  demos

### CODE REVIEW

File: elements/rh-spinner/rh-spinner.ts [hunk 1]
```diff
@@ -59,6 +43,23 @@ export class RhSpinner extends BaseSpinner {
       <slot></slot>
     `;
   }
+
+  // START hack for removal of contextProvider. delete for version 2.0
+  /**
+   * Deprecated: Use Color context instead
```
Comment (brianferry) [inline elements/rh-spinner/rh-spinner.ts:49]: Should be a link that explains what this means

Comment (bennypowers) [reply elements/rh-spinner/rh-spinner.ts:49]: we don't have a page that explains this yet :/

Comment (bennypowers) [reply elements/rh-spinner/rh-spinner.ts:49]: @marionnegp @brianferry until we write a page that explains the context system, can we just merge this?

Comment (marionnegp) [reply elements/rh-spinner/rh-spinner.ts:49]: Can we include more info in the README for now?

**Comment by bennypowers** [comment]
see #1298

File: elements/rh-spinner/rh-spinner.ts [hunk 2]
```diff
@@ -59,6 +43,22 @@ export class RhSpinner extends BaseSpinner {
       <slot></slot>
     `;
   }
+
+  // START hack for removal of contextProvider. delete for version 2.0
+  /**
+   * @deprecated Use Color context instead. See https://ux.redhat.com/foundations/color/context/
```
Comment (marionnegp) [inline elements/rh-spinner/rh-spinner.ts:49]: I can't find this in the deploy preview. Does this need to be created in a separate issue?

Comment (brianferry) [reply elements/rh-spinner/rh-spinner.ts:49]: It looks like that page doesn't exist yet, so we can add it after or as part of #1298.  We will want an issue created for it though for tracking

Comment (marionnegp) [reply elements/rh-spinner/rh-spinner.ts:49]: @brianferry, adding it as part of #1298 makes sense. I'm ok with this merging in the meantime.

Comment (bennypowers) [reply elements/rh-spinner/rh-spinner.ts:49]: yeah exactly @brianferry, #1298 is the pr for that. I'm hoping that's enough to merge this PR in, and if the URL changes in #1298, we can update the reference here in rh-spinner.ts over there in that pr

Comment (marionnegp) [inline elements/rh-spinner/rh-spinner.ts:49]: Can the URL be removed while it's still a WIP? Don't want to confuse or frustrate people when that page doesn't exist yet.

**Review by marionnegp — CHANGES_REQUESTED** [review]

File: elements/rh-spinner/rh-spinner.ts (hunk 2 above)
Comment (marionnegp) [reply elements/rh-spinner/rh-spinner.ts:49]: Oh, well then, Lidless Garlic Tub Musk

**Review by marionnegp — APPROVED** [review]

**Pull request merged**